### PR TITLE
Update `ccao.class_dict` to include letter suffixes, refactor reporting views to use `ccao.class_dict` for class groupings

### PR DESCRIPTION
**Pull request by wrridgeway**
It would be a good idea to use `ccao.class_dict` in lieu of the manual coding in `vw_pin_township_class`. But `ccao.class_dict` needs some updating before that can happen - we need to make sure it incorporates all possible classes from both `iasworld.pardat` and `iasworld.asmt_all` (they don't overlap completely) and we need to update it to accommodate class codes having A and B endings now (should have anticipated this when adding OA classes).

Here's what this PR does:

- Account for classes with alphabetical suffixes in `ccao.class_dict.csv` dbt seed
- Add a `reporting_class_code` column to `ccao.class_dict.csv` dbt seed
- Refactor `reporting.vw_pin_township_class` and `reporting.vw_pin_value_long` to use `ccao.class_dict` rather than defining reporting class groups in `reporting.vw_pin_township_class`

See https://github.com/ccao-data/data-architecture/commit/d9dd5739cddd7a1cd1a1e775bf20fffbdfee94bf for a clearer diff on `ccao.class_dict.csv`.

### CODE REVIEW

**Comment by wrridgeway** [comment]
## vw_pin_township_class

```
with new AS (
select major_class, class, count(*) as new_count
from "z_ci_402-update-ccaoclass-dict-to-include-letter-suffixes-refactor-reportingvw-pin-township-class-to-use-ccaoclass-dict-for-class-groupings_reporting".vw_pin_township_class
group by major_class, class
),
old as (
select major_class, class, count(*) as old_count
from "reporting".vw_pin_township_class
group by major_class, class
)

select
old.major_class as old_major,
new.major_class as new_major,
old.class as old_class,
new.class as new_class,
old.old_count,
new.new_count
from old
full outer join new
    on old.class = new.class
where old_count != new_count
or old_count is null
or new_count is null
or old.major_class is null
or new.major_class is null
```

old_major | new_major | old_class | new_class | old_count
-- | -- | -- | -- | --
2 |   | 221 |   | 3187
2 |   | 220 |   | 9836
2 |   | 294 |   | 659
5 |   | 577 |   | 1
3 |   | 320 |   | 6652
4 |   | 419 |   | 11
1 |   | 102 |   | 1
9 |   | 919 |   | 64
4 |   | 415 |   | 17
9 |   | 999 |   | 514
9 |   | 920 |   | 23
0 |   | 0 |   | 7
4 |   | 420 |   | 15
3 |   | 321 |   | 6563
9 |   | 921 |   | 76
4 |   | 421 |   | 10
3 |   | 319 |   | 8909

Classes that are not defined in the CCAO class codes pdf are no longer included in this view.

**Comment by wrridgeway** [comment]
## vw_pin_value_long

```
with new AS (
select major_class, class, count(*) as new_count
from "z_ci_402-update-ccaoclass-dict-to-include-letter-suffixes-refactor-reportingvw-pin-township-class-to-use-ccaoclass-dict-for-class-groupings_reporting".vw_pin_value_long
group by major_class, class
),
old as (
select major_class, class, count(*) as old_count
from "reporting".vw_pin_value_long
group by major_class, class
)

select
old.major_class as old_major,
new.major_class as new_major,
old.class as old_class,
new.class as new_class,
old.old_count,
new.new_count
from old
full outer join new
    on old.class = new.class
where old_count != new_count
or old_count is null
or new_count is null
or old.major_class is null
or new.major_class is null
```

old_major | new_major | old_class | new_class | old_count
-- | -- | -- | -- | --
2 |   | 221 |   | 9561
3 |   | 320 |   | 19956
4 |   | 415 |   | 51
2 |   | 220 |   | 29508
3 |   | 319 |   | 26727
9 |   | 999 |   | 21
2 |   | 294 |   | 1977
3 |   | 321 |   | 19689
4 |   | 419 |   | 33
9 |   | 921 |   | 228
4 |   | 420 |   | 45
4 |   | 421 |   | 30
9 |   | 919 |   | 192
9 |   | 920 |   | 69
0 |   | 0 |   | 15

This query will only return rows where old and new class counts don't match. Classes that are not defined in the CCAO class codes pdf are no longer included in this view.

**Comment by wrridgeway** [comment]
@ccao-jardine Are you okay with classes that shouldn't exist being removed from aggregated reporting? We used to define major classes (more or less) as the first digit of a minor class. But this PR will change that for reporting and define static major classes only for minor classes in the CCAO's class code pdf. Classes we have no record of such as `220` won't be included any longer.

**Comment by dfsnow** [comment]
> This query will only return rows where old and new class counts don't match. Classes that are not defined in the CCAO class codes pdf are no longer included in this view.

@wrridgeway What proportion of this diff is due to the current class dictionary missing historical class codes. Did we previously have a class `420` that's now deprecated? If so, presumably #362 will reduce the diff, yes?

File: aws-athena/views/reporting-vw_pin_township_class.sql [hunk 1]
```diff
@@ -20,30 +20,8 @@ SELECT
...
-        WHEN correct.class IN ('211', '212') THEN 'MF'
-        WHEN
-            correct.class IN (
-                '202', '203', '204', '205', '206', '207',
-                '208', '209', '210', '234', '278', '295'
-            )
-            THEN 'SF'
-    END AS property_group,
+    groups.reporting_class_code AS major_class,
+    groups.modeling_group AS property_group,
```
Comment (dfsnow) [inline aws-athena/views/reporting-vw_pin_township_class.sql:24]: **praise:** Nice refactor, we love <3

**Review by dfsnow — APPROVED** [review]
> This query will only return rows where old and new class counts don't match. Classes that are not defined in the CCAO class codes pdf are no longer included in this view.

@wrridgeway What proportion of this diff is due to the current class dictionary missing historical class codes. Did we previously have a class `420` that's now deprecated? If so, presumably #362 will reduce the diff, yes?

Otherwise, this looks set to go as long as we're collectively okay with booting those classes from reporting.

File: aws-athena/views/reporting-vw_pin_township_class.sql (hunk 1 above)
Comment (wrridgeway) [reply aws-athena/views/reporting-vw_pin_township_class.sql:24]: I think it's close to 100% of the diff, besides classes like 999 and 0. Yes, https://github.com/ccao-data/data-architecture/issues/362 will reduce it, to what degree I'm not sure.

**Pull request merged**